### PR TITLE
Allow filters on null values

### PR DESCRIPTION
**Pull request by eldadfux**

### CODE REVIEW

File: src/Database/Database.php [hunk 1]
```diff
@@ -1276,17 +1282,19 @@ protected function encodeAttribute(string $name, $value)
      * 
      * @param string $name
      * @param mixed $value
+     * @param Document $document
+     * @param Document $collection
      * 
      * @return mixed
      */
-    protected function decodeAttribute(string $name, $value)
+    protected function decodeAttribute(string $name, $value, Document $document, Document $collection)
```
Comment (kodumbeats) [inline src/Database/Database.php:1290]: Now that we've extended this method to allow us to aggregate&cache the `$collection` document, I think this method could use some additional documentation. Without the Appwrite context, I wouldn't know why we're passing the collection and document.

File: src/Database/Database.php [hunk 2]
```diff
@@ -862,7 +858,7 @@ public function getDocument(string $collection, string $id): Document
 
         $document = $this->casting($collection, $document);
         $document = $this->decode($collection, $document);
-
+
```
Comment (kodumbeats) [inline src/Database/Database.php:861]:
```suggestion
```

**Review by kodumbeats — CHANGES_REQUESTED** [review]

File: src/Database/Database.php (hunk 1 above)
Comment (eldadfux) [reply src/Database/Database.php:1077]: Added a description in the doc-block. Removed the collection for now as we don't really use it at this stage.

**Review by kodumbeats — APPROVED** [review]
👍🏻

**Pull request merged**